frontend/packages/dev-console/src/components/pipelineruns/events/event-utils.ts

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import {
+  WatchK8sResources,
+  useK8sWatchResources,
+  WatchK8sResults,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { PodModel } from '@console/internal/models';
+import {
+  EventInvolvedObject,
+  K8sResourceKind,
+  referenceForModel,
+} from '@console/internal/module/k8s';
+import { PipelineRun, TaskRunKind } from '../../../utils/pipeline-augment';
+import { PipelineRunModel, TaskRunModel } from '../../../models';
+import { PipelineResourceLabel } from '../../pipelines/const';
+
+export type EventFilter = (eventObj: EventInvolvedObject) => boolean;
+type ResourcesType = { [key: string]: K8sResourceKind[] };
+
+/**
+ * Get associated taskruns and pods for a given namespace.
+ * @param namespace
+ */
+export const usePipelineRunRelatedResources = (
+  namespace: string,
+  pipelineRunName: string,
+): WatchK8sResults<ResourcesType> => {
+  const plrRelatedResources: WatchK8sResources<ResourcesType> = React.useMemo(() => {
+    return {
+      taskruns: {
+        kind: referenceForModel(TaskRunModel),
+        namespace,
+        selector: {
+          matchLabels: { [PipelineResourceLabel.pipelinerun]: pipelineRunName },
+        },
+        isList: true,
+        optional: true,
+      },
+      pods: {
+        kind: PodModel.kind,
+        selector: {
+          matchLabels: { [PipelineResourceLabel.pipelinerun]: pipelineRunName },
+        },
+        namespace,
+        isList: true,
+        optional: true,
+      },
+    };
+  }, [namespace, pipelineRunName]);
+  return useK8sWatchResources<ResourcesType>(plrRelatedResources);
+};
+
+const isTaskRunMatched = (taskruns): EventFilter => (taskRun: EventInvolvedObject): boolean =>
+  taskRun.kind === TaskRunModel.kind && taskruns.data.find((t) => t.metadata.uid === taskRun.uid);
+
+const isPodMatched = (pods): EventFilter => (pod: EventInvolvedObject): boolean =>
+  pod.kind === PodModel.kind && pods.data.find((p) => p.metadata.uid === pod.uid);
+
+/**
+ * Custom hook to return the list of event filters to find the pipelinerun related resources.
+ * @param namespace
+ * @param pipelineRun
+ */
+export const usePipelineRunFilters = (
+  namespace: string,
+  pipelineRun: PipelineRun,
+): EventFilter[] => {
+  const { taskruns, pods } = usePipelineRunRelatedResources(namespace, pipelineRun.metadata.name);
+  const isPipelineRunMatched: EventFilter = (plr: EventInvolvedObject): boolean => {
+    return plr.kind === PipelineRunModel.kind && plr.uid === pipelineRun.metadata.uid;
+  };
+  return [isPipelineRunMatched, isTaskRunMatched(taskruns), isPodMatched(pods)];
+};
+
+/**
+ * Custom hook to return the list of event filters to find the taskrun related resources.
+ * @param namespace
+ * @param taskRun
+ */
+export const useTaskRunFilters = (namespace: string, taskRun: TaskRunKind): EventFilter[] => {
+  const pipelineRunResources = usePipelineRunRelatedResources(
+    namespace,
+    taskRun.metadata?.ownerReferences[0].name,
```

Comment:
@karthikjeeyar I created some taskrun resources and some of them didn't have `ownerReferences` property. for those TR resources, the Events tab threw an error while loading. we must handle this scenario. May be need to look for another way to fetch the associated PipelineRun name

I referred this tutorial for creating the resources https://github.com/tektoncd/pipeline/blob/master/docs/tutorial.md

Reply:
@debsmita1 Thanks for pointing out. Absolutely we need to support the taskRuns that are executed directly without pipeline as well. I have handled that case, PTAL.